src/wallet/feebumper.cpp

```diff
@@ -215,7 +215,7 @@ Result CreateRateBumpTransaction(CWallet& wallet, const uint256& txid, const CCo
     // We cannot source new unconfirmed inputs(bip125 rule 2)
     new_coin_control.m_min_depth = 1;
 
-    CTransactionRef tx_new = MakeTransactionRef();
+    CTransactionRef tx_new;
```

Comment:
It appears that the only functional change is that this is initialized to a NULL pointer now, instead of a reference to a dummy transaction. I wonder why this choice was made initially and whether this has any consequences.

Reply:
`CWallet::CreateTransactionInternal` will initialize the transaction. If that fails we will return here early with `return Result::WALLET_ERROR`.

I can add an `Assert(tx_new)`, but I think it is not needed.

Reply:
I'm confused by the `CWallet::CreateTransaction` code here. It makes a copy of the `tx` that is passed in:
```
CTransactionRef tx2 = tx;
```
If it's only an output paramter, why is it copied?

Reply:
Looks like a typo. It should be possible to simply remove the useless assignment with

```diff
diff --git a/src/wallet/wallet.cpp b/src/wallet/wallet.cpp
index 65b54f39b4..d27aba7c7a 100644
--- a/src/wallet/wallet.cpp
+++ b/src/wallet/wallet.cpp
@@ -3109,13 +3109,13 @@ bool CWallet::CreateTransaction(
         bool sign)
 {
     int nChangePosIn = nChangePosInOut;
-    CTransactionRef tx2 = tx;
     bool res = CreateTransactionInternal(vecSend, tx, nFeeRet, nChangePosInOut, error, coin_control, fee_calc_out, sign);
     // try with avoidpartialspends unless it's enabled already
     if (res && nFeeRet > 0 /* 0 means non-functional fee rate estimation */ && m_max_aps_fee > -1 && !coin_control.m_avoid_partial_spends) {
         CCoinControl tmp_cc = coin_control;
         tmp_cc.m_avoid_partial_spends = true;
         CAmount nFeeRet2;
+        CTransactionRef tx2;
         int nChangePosInOut2 = nChangePosIn;
         bilingual_str error2; // fired and forgotten; if an error occurs, we discard the results
         if (CreateTransactionInternal(vecSend, tx2, nFeeRet2, nChangePosInOut2, error2, tmp_cc, fee_calc_out, sign)) {

Reply:
Sounds fair enough. But we also have a changepos in-out which seems coupled, so it is passing in a transaction in some cases?
@meshcollider @achow101 can you weigh in what is supposed to be happening here please.

Reply:
changepos relates to vecSend to shuffle in the change output

Reply:
Pushed a commit to fix the assignment typo

Reply:
Also, added a todo for https://github.com/bitcoin/bitcoin/pull/20588#issuecomment-739838428

Reply:
`CreateTransaction` needs that copy so that the second run of `CreateTransactionInternal` is not polluted by the first run.

Reply:
`tx2` is needed so that the second `CreateTransactionInternal` isn't polluted by the results of the first `CreateTransactionInternal`. But `tx` itself isn't used within `CreateTransactionInternal` at all, so it should be fine to remove the copy.

Reply:
ok, closing this discussion. Let me know if there is anything else you want me to address